<?xml version="1.0" encoding="ISO-8859-1"?>
<launch>
  <arg name="device" default="0" />
  <arg name="use_usb_cam" default="true" />
  <arg name="width" default="640" />
  <arg name="height" default="480" />

  <node pkg="image_publisher" type="image_publisher"
    name="image_pub"
    args="$(find vimjay)/data/clouds.png"
    unless="$(arg use_usb_cam)" >
    <param name="publish_rate" value="5.0"/>
  </node>

  <node name="image_source" pkg="usb_cam" type="usb_cam_node"
      if="$(arg use_usb_cam)"
      output="screen">
    <param name="image_width" value="$(arg width)"/>
    <param name="image_height" value="$(arg height)"/>
    <param name="framerate" value="30"/>
    <param name="video_device" value="/dev/video$(arg device)" />
    <!--param name="pixel_format" value="mjpeg" /-->
    <param name="pixel_format" value="yuyv"/>
  </node>
  <group ns="image_source/image_raw">
    <rosparam param="disable_pub_plugins">
      - 'image_transport/compressed'
      - 'image_transport/compressedDepth'
      - 'image_transport/theora'
    </rosparam>
  </group>

</launch>
